resources/qml/PrintMonitor.qml

```diff
@@ -20,49 +20,462 @@ Column
+                        font: UM.Theme.getFont("default")
+                        color: UM.Theme.getColor("text")
+                        anchors.left: materialColor.right
+                        anchors.leftMargin: UM.Theme.getSize("setting_unit_margin").width
+                        anchors.bottom: parent.bottom
+                        anchors.bottomMargin: UM.Theme.getSize("default_margin").height
+                    }
+                    Text //Variant name.
+                    {
+                        text: printerConnected ? connectedPrinter.hotendIds[index] : ""
+                        font: UM.Theme.getFont("default")
+                        color: UM.Theme.getColor("text")
+                        anchors.right: parent.right
+                        anchors.rightMargin: UM.Theme.getSize("default_margin").width
+                        anchors.bottom: parent.bottom
+                        anchors.bottomMargin: UM.Theme.getSize("default_margin").height
+                    }
+                }
+            }
         }
     }
-    Repeater
+
+    Rectangle
     {
-        model: machineHeatedBed.properties.value == "True" ? 1 : 0
-        delegate: Loader
+        color: UM.Theme.getColor("sidebar")
+        width: parent.width
+        height: machineHeatedBed.properties.value == "True" ? UM.Theme.getSize("sidebar_extruder_box").height : 0
+        visible: machineHeatedBed.properties.value == "True"
+
+        Label //Build plate label.
+        {
+            text: catalog.i18nc("@label", "Build plate")
+            font: UM.Theme.getFont("default")
+            color: UM.Theme.getColor("text")
+            anchors.left: parent.left
+            anchors.leftMargin: UM.Theme.getSize("default_margin").width
+            anchors.top: parent.top
+            anchors.topMargin: UM.Theme.getSize("default_margin").height
+        }
+        Text //Target temperature.
+        {
+            id: bedTargetTemperature
+            text: printerConnected ? connectedPrinter.targetBedTemperature + "°C" : ""
+            font: UM.Theme.getFont("small")
+            color: UM.Theme.getColor("text_inactive")
+            anchors.right: parent.right
+            anchors.rightMargin: UM.Theme.getSize("default_margin").width
+            anchors.bottom: bedCurrentTemperature.bottom
+        }
+        Text //Current temperature.
+        {
+            id: bedCurrentTemperature
+            text: printerConnected ? connectedPrinter.bedTemperature + "°C" : ""
+            font: UM.Theme.getFont("large")
+            color: UM.Theme.getColor("text")
+            anchors.right: bedTargetTemperature.left
+            anchors.rightMargin: UM.Theme.getSize("setting_unit_margin").width
+            anchors.top: parent.top
+            anchors.topMargin: UM.Theme.getSize("default_margin").height
+        }
+        Rectangle //Input field for pre-heat temperature.
```

Comment:
It should in theory be possible to access SettingTextField from here and include it, rather than copying everything over. However, you might need to link a bunch of properties. Just something to consider, not blocking.